### PR TITLE
Action cleanup part 2: Devices

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Tests for devices
- [x] Cleaned up actions
- [x] New unified views for devices

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/devices/device-page/index.stories.js [hunk 1]
```diff
@@ -1,53 +1,37 @@
...
-    }[props.type],
-    revokedAt: props.revoked ? new Date('2002-10-11T01:23:45') : null,
-    showRevokeDevicePage: props.revoked ? null : action('onrevoke'),
-    timeline: [
-      {desc: 'Revoked whenever', subDesc: 'whomever', type: 'Revoked'},
-      {desc: 'Last used whenever', subDesc: 'whenever', type: 'LastUsed'},
-      {desc: 'Added whenever', subDesc: 'provisioner', type: 'Added'},
-    ].slice(0, props.timeLen || 0),
-    type: props.type,
+    showRevokeDevicePage: revoked ? null : action('onrevoke'),
```
Comment (buoyad) [inline shared/devices/device-page/index.stories.js]: :sheep: `action('showRevokeDevicePage')`

File: shared/devices/index.js [hunk 2]
```diff
@@ -1,182 +1,151 @@
...
     return (
-      <Box2 direction="vertical" fullHeight={true} fullWidth={true}>
+      <Common.Box2 direction="vertical" fullHeight={true} fullWidth={true}>
         <DeviceHeader
           setAttachmentRef={this.props.setAttachmentRef}
           onAddNew={this.props.toggleShowingMenu}
           waiting={this.props.waiting}
         />
-        <List items={items} renderItem={this._renderRow} />
+        {this.props.waiting && <Common.ProgressIndicator style={styles2.progress} />}
```
Comment (buoyad) [inline shared/devices/index.js]: This is rendering in the upper left corner of the window, on top of the tab bar
image

File: shared/devices/device-revoke/index.js [hunk 3]
```diff
@@ -1,208 +1,140 @@
...
+        ) : (
+          <Kb.Text type="Header" style={styles.italicName}>
+            {props.device.name}
+          </Kb.Text>
+        )}
+        ?
+      </Kb.Text>
+      <Kb.Box2 direction="vertical" style={styles.endangeredTLFContainer} fullWidth={Styles.isMobile}>
+        {props.waiting ? (
+          <Kb.ProgressIndicator />
```
Comment (buoyad) [inline shared/devices/device-revoke/index.js]: Could this be positioned absolutely somewhere? it makes things quickly jump around

Comment (chrisnojima) [reply shared/devices/device-revoke/index.js]: i'm removing it and adding text to the bottom
@cecileboucheron 
screen shot 2018-07-31 at 10 53 07 am

**Pull request merged**